We planned once to add a debioan icon in the titlebar of active windows.
Unused right now.
--- a/kwin/clients/oxygen/oxygenclient.cpp
+++ b/kwin/clients/oxygen/oxygenclient.cpp
@@ -57,6 +57,9 @@
 #include "oxygenbutton.h"
 #include "oxygen.h"
 
+#define logourl "/usr/share/kde4/apps/kwin/titlebar_decor.png"
+#define logoOffset 9
+
 namespace Oxygen
 {
 
@@ -73,6 +76,7 @@
     , colorCacheInvalid_(true)
     , helper_(*globalHelper)
 {
+    logo.load(logourl);
 }
 
 OxygenClient::~OxygenClient()
@@ -237,9 +241,9 @@
 }
 
 
-QColor OxygenClient::titlebarTextColor(const QPalette &palette)
+QColor OxygenClient::titlebarTextColor(const QPalette &palette, bool active)
 {
-    if (isActive())
+    if (active)
         return palette.color(QPalette::Active, QPalette::WindowText);
     else {
         if(colorCacheInvalid_) {
@@ -255,6 +259,9 @@
     }
 }
 
+QColor OxygenClient::titlebarTextColor(const QPalette &palette) {
+    return titlebarTextColor(palette, isActive());
+}
 
 void OxygenClient::paintEvent(QPaintEvent *e)
 {
@@ -280,12 +287,14 @@
     const int titleEdgeLeft = layoutMetric(LM_TitleEdgeLeft);
     const int marginLeft = layoutMetric(LM_TitleBorderLeft);
     const int marginRight = layoutMetric(LM_TitleBorderRight);
+    const int borderLeft = layoutMetric(LM_BorderLeft, true) + (layoutMetric(LM_BorderLeft, true) == 0 ? -1 : 0);
 
     const int titleLeft = frame.left() + titleEdgeLeft + buttonsLeftWidth() + marginLeft;
+    const int fullLogoOffset = (logo.isNull() ? 0 : logo.width() + logoOffset - borderLeft);
     const int titleWidth = frame.width() -
             titleEdgeLeft - layoutMetric(LM_TitleEdgeRight) -
             buttonsLeftWidth() - buttonsRightWidth() -
-            marginLeft - marginRight;
+            marginLeft - marginRight - fullLogoOffset;
 
 
     int splitY = qMin(300, 3*frame.height()/4);
@@ -317,6 +326,15 @@
     painter.drawText(titleLeft, titleTop-1, titleWidth, titleHeight,  // -1 is to go into top resizearea
               OxygenFactory::titleAlign() | Qt::AlignVCenter, caption());
 
+    if (!logo.isNull() && isActive()) {
+        QPixmap pix(logo.width(),logo.height());
+        pix.fill(titlebarTextColor(palette, false));
+        pix.setAlphaChannel(logo.alphaChannel());
+        QFontMetrics fm(options()->font(isActive(), false));
+        painter.drawPixmap(titleEdgeLeft + layoutMetric(LM_TitleEdgeRight) + buttonsLeftWidth()
+          + qMin(fm.width(caption()), titleWidth) + logoOffset - borderLeft, titleTop, pix);
+    }
+
     painter.setRenderHint(QPainter::Antialiasing);
 
     // Draw dividing line
--- a/kwin/clients/oxygen/oxygenclient.h
+++ b/kwin/clients/oxygen/oxygenclient.h
@@ -53,9 +53,11 @@
 
 private:
     void paintEvent(QPaintEvent *e);
+    QColor titlebarTextColor(const QPalette &palette, bool active);
     QColor titlebarTextColor(const QPalette &palette);
     bool colorCacheInvalid_;
     QColor cachedTitlebarTextColor_;
+    QPixmap logo;
 
 protected:
     friend class OxygenButton;
